client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -89,9 +91,19 @@ tie.factory('PythonCodeRunnerService', [
         }
 
         // The run was successful.
-        return CodeEvalResultObjectFactory.create(
+        var codeEvalResult = CodeEvalResultObjectFactory.create(
           code, outputLines.join('\n'), observedOutputs,
           buggyOutputTestResults, performanceTestResults, null, null);
+        var question = CurrentQuestionService.getCurrentQuestion();
+        if (!question) {
```

Comment:
Why would there be no question?

Reply:
Fixed specs to accurately always have a question.